### PR TITLE
Enable cloudbuild to receive Github events.

### PR DESCRIPTION
**Pull request by ndmckinley**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
Support `github` field in `google_cloudbuild_trigger`.
```

### CODE REVIEW

File: templates/terraform/examples/cloudbuild_trigger_github.tf.erb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+resource "google_cloudbuild_trigger" "<%= ctx[:primary_resource_id] %>" {
+  trigger_template {
+    branch_name = "master"
+    repo_name   = "my-repo"
```
Comment (naseemkullah) [inline templates/terraform/examples/cloudbuild_trigger_github.tf.erb]: I think a trigger can either have `trigger_template` or `github` object but not both. Please double check this.

File: templates/terraform/examples/cloudbuild_trigger_github.tf.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  substitutions = {
+    _FOO = "bar"
+    _BAZ = "qux"
+  }
+
+  github {
+    installation_id = "abc123"
```
Comment (naseemkullah) [inline templates/terraform/examples/cloudbuild_trigger_github.tf.erb]: Will this be auto populated or must we specify an `installation_id` ?

File: templates/terraform/examples/cloudbuild_trigger_github.tf.erb (hunk 1 above)
Comment (ndmckinley) [reply templates/terraform/examples/cloudbuild_trigger_github.tf.erb]: In general, the docs will say which fields are exclusive, and they don't on this one, but I'll take a look.

File: templates/terraform/examples/cloudbuild_trigger_github.tf.erb (hunk 2 above)
Comment (ndmckinley) [reply templates/terraform/examples/cloudbuild_trigger_github.tf.erb]: Yes, that will be a required field.

File: templates/terraform/examples/cloudbuild_trigger_github.tf.erb (hunk 1 above)
Comment (wlynch) [reply templates/terraform/examples/cloudbuild_trigger_github.tf.erb]: Correct! Only one of these can be set at a time. I'll make sure the API docs get updated to reflect this.

File: templates/terraform/examples/cloudbuild_trigger_github.tf.erb (hunk 2 above)
Comment (wlynch) [reply templates/terraform/examples/cloudbuild_trigger_github.tf.erb]: This field is actually ignored, but was missed for cleanup prior to the beta release. Feel free to ignore this field. I'll make sure this gets marked as deprecated and eventually cleaned up.

Comment (ndmckinley) [reply templates/terraform/examples/cloudbuild_trigger_github.tf.erb]: Ah, that's wonderful.  Thanks!  I'll remove it.

**Comment by ndmckinley** [comment]
Whoops, can't add that documentation here since that field isn't in the other providers.

**Pull request merged**